### PR TITLE
Use credo

### PR DESCRIPTION
**Pull request by davydog187**
Added credo for keeping the code looking great. Turned off the check for PipeStartChain and Spec, and lowered the priority for paramater matching (it was incorrectly reporting).

### CODE REVIEW

**Comment by davydog187** [comment]
I wrote some placeholder moduledocs. Would be happy to replace them with whatever.

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@
 /doc
 erl_crash.dump
 *.ez
+$TMPDIR/
```
Comment (BlakeWilliams) [inline .gitignore]: Can we remove this?

File: lib/mix/tasks/update_slack_api.ex [hunk 2]
```diff
@@ -1,19 +1,21 @@
 defmodule Mix.Tasks.UpdateSlackApi do
+  @moduledoc "Task to update the Slack API documentation files"
```
Comment (BlakeWilliams) [inline lib/mix/tasks/update_slack_api.ex]: `Updates Slack API documentation files for generating API code`

File: lib/slack/bot.ex [hunk 3]
```diff
@@ -76,7 +76,7 @@ defmodule Slack.Bot do
   end
 
   @doc false
-  def onconnect(_websocket_request, %{slack: slack, process_state: process_state, bot_handler: bot_handler} = state) do
+  def onconnect(_websocket_request, state = %{slack: slack, process_state: process_state, bot_handler: bot_handler} = state) do
```
Comment (BlakeWilliams) [inline lib/slack/bot.ex]: Can we remove the `= state` at the end?

File: mix.exs [hunk 4]
```diff
@@ -21,8 +21,11 @@ defmodule Slack.Mixfile do
     [{:httpoison, "~> 0.11"},
      {:exjsx, "~> 3.2.0"},
      {:websocket_client, "~> 1.1.0"},
+
```
Comment (BlakeWilliams) [inline mix.exs]: Remove newline please

File: mix.exs [hunk 5]
```diff
@@ -21,8 +21,11 @@ defmodule Slack.Mixfile do
...
      {:earmark, "~> 0.2.0", only: :dev},
-     {:ex_doc, "~> 0.12", only: :dev}]
+     {:ex_doc, "~> 0.12", only: :dev},
+     {:credo, "~> 0.5", only: [:dev, :test]}
```
Comment (BlakeWilliams) [inline mix.exs]: Do we need `:test` too?

File: lib/slack/bot.ex (hunk 3 above)
Comment (davydog187) [inline lib/slack/bot.ex]: I was trying to figure out why credo was reporting the other way as inconsistent, and forgot to change it back. Fixing!

File: mix.exs (hunk 5 above)
Comment (davydog187) [inline mix.exs]: Yes, if you want to run it as part of the CI

File: mix.exs (hunk 4 above)
Comment (davydog187) [inline mix.exs]: Thats typically done to visually separate dev/prod dependencies. I'll remove if thats what you prefer

**Comment by davydog187** [comment]
@BlakeWilliams addressed your feedback.

**Comment by BlakeWilliams** [comment]
Nice, this is a great addition. Thanks!

**Pull request merged**